lib/harvesting/client.rb

```diff
@@ -148,6 +148,9 @@ def http_response(method, uri, opts = {})
 
       raise Harvesting::AuthenticationError.new(response.to_s) if auth_error?(response)
       raise Harvesting::UnprocessableRequest.new(response.to_s) if response.code.to_i == 422
+      if response.code.to_i == 404
+        raise Harvesting::RequestNotFound.new('The page you were looking for may have been moved or the address misspelled.')
+      end
```

Comment:
@JuannFerrari I think maybe we can add a bit more value to the user if we tell them what URI they were trying to send a request to? 

```suggestion
      raise Harvesting::RequestNotFound.new(uri) if response.code.to_i == 404
```